src/audio/audio.c

```diff
@@ -94,69 +99,86 @@ void audio_get_devices()
 * @param y (dx)
 * @param z (bp)
 */
-int sound_play_panned(int sound_id, int ebx, sint16 x, sint16 y, sint16 z)
+int audio_sound_play_panned(int soundId, int ebx, sint16 x, sint16 y, sint16 z)
 {
-	if (!gGameSoundsOff) {
-		int volumedown = 0;
-		int volume = 0;
-		if (ebx == 0x8001) {
-			rct_map_element* mapelement = map_get_surface_element_at(x / 32, y / 32);
-			if (mapelement) {
-				if ((mapelement->base_height * 8) - 5 > z) {
-					volumedown = 10;
-				}
-			}
-			sint16 rx;
-			sint16 ry;
-			switch (get_current_rotation()) {
-				case 0:
-					rx = y - x;
-					ry = ((y + x) / 2) - z;
-					break;
-				case 1:
-					rx = -x - y;
-					ry = ((y - x) / 2) - z;
-					break;
-				case 2:
-					rx = x - y;
-					ry = ((-y - x) / 2) - z;
-					break;
-				case 3:
-					rx = y + x;
-					ry = ((x - y) / 2) - z;
-					break;
-			}
-			rct_window* window = RCT2_GLOBAL(RCT2_ADDRESS_NEW_WINDOW_PTR, rct_window*);
-			while (1) {
-				window--;
-				if (window < RCT2_ADDRESS(RCT2_ADDRESS_WINDOW_LIST, rct_window)) {
-					break;
-				}
-				rct_viewport* viewport = window->viewport;
-				if (viewport && viewport->flags & VIEWPORT_FLAG_SOUND_ON) {
-					sint16 vy = ry - viewport->view_y;
-					sint16 vx = rx - viewport->view_x;
-					ebx = viewport->x + (vx >> viewport->zoom);
-					volume = RCT2_ADDRESS(0x0099282C, int)[sound_id] + ((-1024 * viewport->zoom - 1) << volumedown) + 1;
-					if (vy < 0 || vy >= viewport->view_height || vx < 0 || vx >= viewport->view_width || volume < -10000) {
-						return sound_id;
-					}
-				}
-			}
+	if (gGameSoundsOff)
+		return 0;
+
+	int volume = 0;
+	if (ebx == 0x8001) {
```

Comment:
ebx is the pan with a range from 0x8000 (-32768) to 7FFF (32767).
It has two special values, 0x8000 and 0x8001 which should be defined as named constants.

Its not particularly nice, but we have to keep it like this until we know we have implemented all play sound callers.